src/NUnitFramework/framework/Internal/Randomizer.cs

```diff
@@ -563,18 +563,67 @@ public string GetString()
         /// </summary>
         public decimal NextDecimal()
         {
-            int low = Next(0, int.MaxValue);
-            int mid = Next(0, int.MaxValue);
-            int high = Next(0, int.MaxValue);
-            return new Decimal(low, mid, high, false, 0);
+            var bytes = new byte[sizeof(int) * 3];
+            NextBytes(bytes);
+
+            var low = BitConverter.ToInt32(bytes, 0);
+            var mid = BitConverter.ToInt32(bytes, 4);
+            var high = BitConverter.ToInt32(bytes, 8);
+
+            return new decimal(low, mid, high, false, 0);
         }
 
         /// <summary>
         /// Returns a random decimal between positive zero and the specified maximum.
         /// </summary>
         public decimal NextDecimal(decimal max)
         {
-            return NextDecimal() % max;
```

Comment:
I'm almost ready to bet that we have other uses of modulo with other numeric types. Do we need to test them too?

Reply:
Yes. Should I open a separate issue for non-decimal types rather than do everything in this PR?

Reply:
I'm for small PRs!

Reply:
I just checked and I don't see anything else in Randomizer.cs that is using modulo without also correcting for its bias.